src/components/SearchResultListing/SearchResultButtons.js

```diff
@@ -1,53 +1,169 @@
+              {glyph && '\u00A0'}
+              {i18n.getMessage(titleKey)}
+            </span>
+          }
+          id={uid}
+          key={uid}
+          onClick={onClick}
+          bsSize="sm"
+        >
+          {
+            operations.map(({ name, icon, handler }, index) => (
+              <MenuItem
+                key={index}
+                eventKey={index}
+                onClick={handler}
+              >
+                <span className="btn-sm text-left">
+                  {icon && <Glyphicon glyph={icon}/>}
+                  {icon && '\u00A0\u00A0'}
+                  {getModelMessage(i18n, `model.label.${name}`, name)}
+                </span>
+              </MenuItem>
+            ))
+          }
+        </SplitButton>
+      ) :
+      (
+        <Button onClick={onClick} key={uid}>
+          {glyph && <Glyphicon glyph={glyph} />}
+          {glyph && ' '}
+          {i18n.getMessage(titleKey)}
+        </Button>
+      )
+  }
+
+  makeExternalOpsButtons = _ => {
+    const { operations: allOps, index } = this.props;
+    const uid = `ops-external-${index}`;
+    const { i18n } = this.context;
+
+    const operations = allOps.filter(({ type }) => type === 'external');
+
+    if (operations.length > 1) {
+      const { icon, name, handler } = operations[0];
+
+      return (
+        <SplitButton
+          title={
+            <span>
+              {icon && <Glyphicon glyph={icon}/>}
+              {icon && '\u00A0'}
+              {getModelMessage(i18n, `model.label.${name}`, name)}
+            </span>
+          }
+          id={uid}
+          key={uid}
+          onClick={handler}
+          bsSize="sm"
+        >
+          {
+            operations.slice(1).map(({ name, icon, handler }, index) => (
+              <MenuItem
+                key={index}
+                eventKey={index}
+                onClick={handler}
+              >
+                <span className="btn-sm text-left">
+                  {icon && <Glyphicon glyph={icon}/>}
+                  {icon && '\u00A0\u00A0'}
+                  {getModelMessage(i18n, `model.label.${name}`, name)}
+                </span>
+              </MenuItem>
+            ))
+          }
+        </SplitButton>
+      )
+    } else if (operations.length === 1) {
+      const { icon, name, handler } = operations[0];
+
+      return (
+        <Button onClick={handler} key={uid}>
+          {icon && <Glyphicon glyph={icon} />}
+          {icon && ' '}
+          {getModelMessage(i18n, `model.label.${name}`, name)}
+        </Button>
+      )
+    }
+
+    return null;
+  }
+
   render() {
-    const permissions = this.props.model.data.permissions.crudOperations;
-    const { onShow, onEdit, onDelete } = this.props;
+    const {
+      onShow,
+      onEdit,
+      onDelete,
+      permissions
+    } = this.props;
+
     const { i18n } = this.context;
 
     const buttons = [];
 
     if (permissions.edit) {
```

Comment:
Unite makeExternalOpsButtons() and makeCustomOpsButtons() into one operationButtons(). Replace lines 147-165 with
```javascript
buttons.push(
  this.operationButtons([
    ...(
      permissions.edit ?
        [{
          icon: 'edit',
          name: 'crudEditor.edit.button',
          handler: onEdit
        }] :
        (
          permissions.view ?
            [{
              icon: 'eye-open',
              name: 'crudEditor.show.button',
              handler: onShow
            }] :
            []
        )
    ),
    ...this.props.operations.filter(({ type }) => type === 'custom')
  )
);

buttons.push(
  this.operationButtons(
    this.props.operations.filter(({ type }) => type === 'external')
  )
);
```